src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -121,14 +121,30 @@ internal partial class Binder
             BindingDiagnosticBag diagnostics)
         {
             beforeAttributePartBound?.Invoke(node);
-            var boundAttribute = new ExecutableCodeBinder(node, this.ContainingMemberOrLambda, this).BindAttribute(node, boundAttributeType, (this as ContextualAttributeBinder)?.AttributedMember, diagnostics);
+            var boundAttribute = new ExecutableCodeBinder(node, this.ContainingMemberOrLambda, this).BindAttribute(node, boundAttributeType, (this as ContextualAttributeBinder)?.AttributeTarget, diagnostics);
             afterAttributePartBound?.Invoke(node);
             return (GetAttribute(boundAttribute, diagnostics), boundAttribute);
         }
 
         internal BoundAttribute BindAttribute(AttributeSyntax node, NamedTypeSymbol attributeType, Symbol? attributedMember, BindingDiagnosticBag diagnostics)
         {
-            return this.GetRequiredBinder(node).BindAttributeCore(node, attributeType, attributedMember, diagnostics);
+            return this.GetRequiredBinder(node).BindAttributeCore(node, attributeType, GetAttributedMember(attributedMember), diagnostics);
+
+            static Symbol? GetAttributedMember(Symbol? symbol)
+            {
+                for (; symbol is not null; symbol = symbol.ContainingSymbol)
+                {
+                    switch (symbol.Kind)
+                    {
+                        case SymbolKind.Method:
+                        case SymbolKind.Property:
+                        case SymbolKind.Event:
+                            return symbol;
+                    }
+                }
+
+                return symbol;
+            }
```

Comment:
I had to move this from `ContextualAttributeBinder` so that it gets shared between `AttributeSemanticModel`